src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UsageReport/WriteUsageBurndownData.cs

```diff
@@ -87,8 +88,8 @@ public override bool Execute()
         /// <returns>The contents of the specified file.</returns>
         private string GetFileContents(string relativeFilePath, string commitSha)
         {
-            WebClient client = new WebClient();
-            var xmlString = client.DownloadString($"https://raw.githubusercontent.com/dotnet/source-build/{commitSha}/{relativeFilePath.Replace('\\', '/')}");
+            HttpClient client = new HttpClient();
```

Comment:
I would suggest keeping it as a local variable since it's not needed anywhere else. Also, you can [omit the type name for the `new` keyword](https://learn.microsoft.com/dotnet/csharp/language-reference/operators/new-operator#target-typed-new).

The reason the `using` declaration is needed is because `HttpClient` implements `IDisposable`. It's always a best practice to explicitly dispose of such types. The [`using` declaration](https://learn.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-8.0/using#using-declaration) in this context does that for you.

```suggestion
            using HttpClient client = new();
```

Reply:
> I would suggest keeping it as a local variable since it's not needed anywhere else.

This method is invoked within a [loop](https://github.com/dotnet/installer/blob/383cb48f5a36ff1e6bc2b7cd18058ef18430c6b2/src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UsageReport/WriteUsageBurndownData.cs#L55) however.  Given the guidance to minimize the HttpClient instances per app, pulling this outside the method makes sense IMO.

Reply:
I missed that, thanks. Then, yes, if it's in a loop it should definitely be pulled out to only be instantiated once.

Reply:
Updated [here ](https://github.com/dotnet/installer/pull/17128/commits/c5094a20a7ff08ef2c48d0c425e5441e798c17fc#diff-5f7db611ced72630c248b14f2f95729238e69d5677b4793b15ffe313d568b5f2R50)to only instantiate HttpClient once.